src/main/java/mil/dds/anet/resources/AttachmentResource.java

```diff
@@ -0,0 +1,193 @@
+      // only admin or owner can update attachment
+      throw new WebApplicationException("You don't have permission to delete this attachment",
+          Status.FORBIDDEN);
+    }
+
+    final int numRows = dao.delete(attachmentUuid);
+    if (numRows == 0) {
+      throw new WebApplicationException("Couldn't process attachment delete", Status.NOT_FOUND);
+    }
+    AnetAuditLogger.log("Attachment {} deleted by {}", attachmentUuid, user);
+    return numRows;
+  }
+
+  @GET
+  @Timed
+  @Path("/download/{uuid}")
+  @Produces(MediaType.APPLICATION_OCTET_STREAM)
+  public Response downloadAttachment(final @Auth Person user, @PathParam("uuid") String uuid) {
+    final Attachment attachment = getAttachment(uuid);
+    final ResponseBuilder response =
+        Response.ok(streamContentBlob(uuid)).type(MediaType.APPLICATION_OCTET_STREAM)
+            .header("Content-Disposition", getContentDisposition("attachment", attachment));
+    return response.build();
+  }
+
+  @GET
+  @Timed
+  @Path("/view/{uuid}")
+  public Response viewAttachment(final @Auth Person user, @PathParam("uuid") String uuid) {
+    final Attachment attachment = getAttachment(uuid);
+    final ResponseBuilder response =
+        Response.ok(streamContentBlob(uuid)).type(attachment.getMimeType())
+            .header("Content-Disposition", getContentDisposition("inline", attachment));
+    return response.build();
+  }
+
+  private Attachment getAttachment(final String uuid) {
+    final Attachment attachment = dao.getByUuid(uuid);
+    if (attachment == null) {
+      throw new WebApplicationException("Attachment not found", Status.NOT_FOUND);
+    }
+    return attachment;
+  }
+
+  private StreamingOutput streamContentBlob(final String uuid) {
+    return output -> dao.streamContentBlob(uuid, output);
+  }
+
+  private static String getContentDisposition(String disposition, Attachment attachment) {
+    final ParameterList parameterList = new ParameterList();
+    parameterList.set("filename", attachment.getFileName(), StandardCharsets.UTF_8.toString());
+    return new ContentDisposition(disposition, parameterList).toString();
+  }
+
+  private boolean hasUploadPermission(final Person user, final String mimeType) {
+    final String keyPath = "fields.attachment";
+    final Map<String, Object> uploadPermission =
+        (Map<String, Object>) AnetObjectEngine.getConfiguration().getDictionaryEntry(keyPath);
+    final Boolean userPermission = (Boolean) uploadPermission.get("disabled");
+
+    if (userPermission && !AuthUtils.isAdmin(user)) {
+      throw new WebApplicationException("You don't have permission to upload attachment",
+          Response.Status.FORBIDDEN);
+    }
+
+    final List<String> allowedMimetypes = ((List<String>) uploadPermission.get("mimeType")).stream()
+        .map(String::toString).collect(Collectors.toList());
+    if (!allowedMimetypes.contains(mimeType)) {
+      throw new WebApplicationException("File extension is not allowed",
```

Comment:
Suggest to rephrase to `String.format("Files of type %s are not allowed", mimeType)`.